test/automation/src/sql/notebook.ts

```diff
@@ -78,6 +84,23 @@ export class Notebook {
 		const outputComponent = '.notebook-cell.active .notebook-output';
 		await this.code.waitForElement(outputComponent);
 	}
+
+	async waitForAllResults(cellIds: number[]): Promise<void> {
```

Comment:
When I see a function named something like "waitForAll..." I would assume that it does actually wait for ALL the results which isn't the case here.

I'd suggest : 

1. Renaming this to `waitForResults` with the same parameter
1. Adding a new function `waitForAllResults` that does wait for all results without needing to pass in cell IDs

Reply:
The same instance of ADS is used for each smoke test. If another notebook smoke test is added before this one then it would break this test because of the cell id changes. But it would not break if tests were added after this one.

The issue I ran into while implementing waitForAllResults is that there is no way to identify each cell so I added the cell.id to select each cell. I can change the method name to waitForResults.

In order to implement a waitForAllResults method that doesn't rely on the cell ids, I think we would need some keyboard shortcuts (ex. focus on first cell, focus next cell).

Reply:
For waiting for all results you should be able to get the list of cells by using the selector `div.notebook-cell` and then for each of those elements wait for the child element with the `notebook-output` class.

Why do you think we need keyboard shortcuts?

Reply:
Hm I need to do a little more investigation but I remember running into an issue where using div.notebook-cell will only select the first cell. Let me try again.

Reply:
Well it depends on what you're doing. waitForElement will pick the first one. So make sure you're using `waitForElements` https://github.com/microsoft/azuredatastudio/blob/d2bdd2bace223ecc9436eee27b47b2618a353de6/test/automation/src/code.ts#L312

Reply:
That works! Thank you Charles!